partner_tracking/models/res_partner.py

```diff
@@ -0,0 +1,98 @@
+    lang = fields.Selection(track_visibility='onchange')
+    tz = fields.Selection(track_visibility='onchange')
+    user_id = fields.Many2one(track_visibility='onchange')
+    vat = fields.Char(track_visibility='onchange')
+    website = fields.Char(track_visibility='onchange')
+    comment = fields.Text(track_visibility='onchange')
+    credit_limit = fields.Float(track_visibility='onchange')
+    ean13 = fields.Char(track_visibility='onchange')
+    active = fields.Boolean(track_visibility='onchange')
+    customer = fields.Boolean(track_visibility='onchange')
+    supplier = fields.Boolean(track_visibility='onchange')
+    employee = fields.Boolean(track_visibility='onchange')
+    function = fields.Char(track_visibility='onchange')
+    type = fields.Selection(track_visibility='onchange')
+    street = fields.Char(track_visibility='onchange')
+    street2 = fields.Char(track_visibility='onchange')
+    zip = fields.Char(track_visibility='onchange')
+    city = fields.Char(track_visibility='onchange')
+    state_id = fields.Many2one(track_visibility='onchange')
+    country_id = fields.Many2one(track_visibility='onchange')
+    email = fields.Char(track_visibility='onchange')
+    phone = fields.Char(track_visibility='onchange')
+    fax = fields.Char(track_visibility='onchange')
+    mobile = fields.Char(track_visibility='onchange')
+    birthdate = fields.Char(track_visibility='onchange')
+    is_company = fields.Boolean(track_visibility='onchange')
+    use_parent_address = fields.Boolean(track_visibility='onchange')
+    has_image = fields.Boolean(track_visibility='onchange')
+    company_id = fields.Many2one(track_visibility='onchange')
+    color = fields.Integer(track_visibility='onchange')
+
+    @api.multi
+    def action_set_controlled(self):
+        """
+        Change the state to 'controlled'.
+        """
+        for rec in self:
+            rec.state = 'controlled'
+
+    @api.multi
+    def write(self, vals):
+        """
+        Change state to 'pending' if values have been updated by a user that
+        is not part of the validation group
+        """
+        if self.env.context.get('params'):
+            user_preferences_action = self.env.ref('base.action_res_users_my')
+            if (
+                self.env.context.get('params').get('action') ==
+                user_preferences_action.id
+            ):  # write() is called from the user changing his preferences
+                return super(ResPartner, self).write(vals)
+        user = self.env['res.users'].browse(self.env.uid)
+        if not user.has_group('partner_tracking.group_partner_validation'):
+            for rec in self:
+                if rec.state == 'controlled':
+                    tracked_vals = [field for field in vals if (
+                        field in tracked_fields
+                    )]
+                    if any(rec[field] != vals[field] for field in
+                           tracked_vals):
```

Comment:
@dufresnedavid I've posted a question on an outdated diff so I think you haven't seen it:
The module depends on 'account' (because of where is located the menu item). Should I also add the fields defined in account?